File: android/src/main/java/com/rebeloid/unity_ads/UnityAdsConstants.java
package com.rebeloid.unity_ads;

public interface UnityAdsConstants {
    String MAIN_CHANNEL = "com.rebeloid.unity_ads";

    String BANNER_AD_CHANNEL = MAIN_CHANNEL + "/bannerAd";
    String VIDEO_AD_CHANNEL = MAIN_CHANNEL + "/videoAd";

    String PLACEMENT_ID_PARAMETER = "placementId";
    String ERROR_CODE_PARAMETER = "errorCode";
    String ERROR_MESSAGE_PARAMETER = "errorMessage";

    //initialize
    String INIT_METHOD = "init";
    String FIREBASE_TEST_LAB_MODE_PARAMETER = "firebaseTestLabMode";
    String GAME_ID_PARAMETER = "gameId";
    String TEST_MODE_PARAMETER = "testMode";
    String INIT_COMPLETE_METHOD = "initComplete";
    String INIT_FAILED_METHOD = "initFailed";

    //load
    String LOAD_METHOD = "load";
    String LOAD_COMPLETE_METHOD = "loadComplete";
    String LOAD_FAILED_METHOD = "loadFailed";

    //show
    String SHOW_VIDEO_METHOD = "showVideo";
    String SERVER_ID_PARAMETER = "serverId";
    String SHOW_COMPLETE_METHOD = "showComplete";
    String SHOW_FAILED_METHOD = "showFailed";
    String SHOW_START_METHOD = "showStart";
    String SHOW_SKIPPED_METHOD = "showSkipped";
    String SHOW_CLICK_METHOD = "showClick";

    //banner
    String HEIGHT_PARAMETER = "height";
    String WIDTH_PARAMETER = "width";
    String BANNER_ERROR_METHOD = "banner_error";
    String BANNER_LOADED_METHOD = "banner_loaded";
    String BANNER_CLICKED_METHOD = "banner_clicked";

    //privacy consent
    String PRIVACY_CONSENT_SET_METHOD = "privacyConsent_set";
    String PRIVACY_CONSENT_TYPE_PARAMETER = "privacyConsent_type";
    String PRIVACY_CONSENT_VALUE_PARAMETER = "privacyConsent_value";
}

File: ios/Classes/UnityAdsConstants.swift
struct UnityAdsConstants {
    static let MAIN_CHANNEL = "com.rebeloid.unity_ads";
    
    static let BANNER_AD_CHANNEL = MAIN_CHANNEL + "/bannerAd";
    static let VIDEO_AD_CHANNEL = MAIN_CHANNEL + "/videoAd";
    
    static let PLACEMENT_ID_PARAMETER = "placementId";
    static let ERROR_CODE_PARAMETER = "errorCode";
    static let ERROR_MESSAGE_PARAMETER = "errorMessage";
    
    //initialize
    static let INIT_METHOD = "init";
    static let GAME_ID_PARAMETER = "gameId";
    static let TEST_MODE_PARAMETER = "testMode";
    static let INIT_COMPLETE_METHOD = "initComplete";
    static let INIT_FAILED_METHOD = "initFailed";
    static let IS_INITIALIZED_METHOD = "isInitialized";
    
    //load
    static let LOAD_METHOD = "load";
    static let LOAD_COMPLETE_METHOD = "loadComplete";
    static let LOAD_FAILED_METHOD = "loadFailed";
    
    //show
    static let SHOW_VIDEO_METHOD = "showVideo";
    static let SERVER_ID_PARAMETER = "serverId";
    static let SHOW_COMPLETE_METHOD = "showComplete";
    static let SHOW_FAILED_METHOD = "showFailed";
    static let SHOW_START_METHOD = "showStart";
    static let SHOW_SKIPPED_METHOD = "showSkipped";
    static let SHOW_CLICK_METHOD = "showClick";
    
    //banner
    static let HEIGHT_PARAMETER = "height";
    static let WIDTH_PARAMETER = "width";
    static let BANNER_ERROR_METHOD = "banner_error";
    static let BANNER_LOADED_METHOD = "banner_loaded";
    static let BANNER_CLICKED_METHOD = "banner_clicked";
    
    //privacy consent
    static let PRIVACY_CONSENT_SET_METHOD = "privacyConsent_set";
    static let PRIVACY_CONSENT_TYPE_PARAMETER = "privacyConsent_type";
    static let PRIVACY_CONSENT_VALUE_PARAMETER = "privacyConsent_value";
}
